src/Gui/Placement.cpp

```diff
@@ -462,10 +467,12 @@ Placement::Placement(QWidget* parent, Qt::WindowFlags fl)
     setupUnits();
     setupSignalMapper();
     setupRotationMethod();
+    setupEventFilter();
 }
 
 Placement::~Placement()
 {
+    if (qApp!=NULL) qApp->removeEventFilter(this);

```

Comment:
This check is not needed. I have never seen qApp being null in a GUI application. Btw, NULL is C and in modern C++ nullptr must be used.

Reply:
Check removed.

Assumed that it mitigated a segmentation fault that I have seen.
Unfortunately triggering it was not very repeatable and this check does not help with that segfault. 
I can also trigger that segfault on current unmodified Freecad:main branch. 

Could you check if you can trigger it by something like this:
1. open any document (lets say "test.FCStd")
2. open placement dialog on any Body in that document
3. move object by spacenavd or mouse scroll wheel on angle spinbox
4. open any other document without closing placement dialog
5. close test.FCStd
6. move object by spacenavd or mouse scroll wheel on angle spinbox (trigger is hard to predict, sometimes value needs to be changed many times per second or enough times to trigger, sometimes a single value change can trigger a segmentation fault, I have seen a higher probability with spacenavd)

After that procedure i can get different segmentation faults.
For example like this:

`
No object selected.
Program received signal SIGSEGV, Segmentation fault.
#0  /lib/x86_64-linux-gnu/libc.so.6(+0x3c050) [0x7fdd3e369050]
#1  0x7fdd3e67e014 in std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::c_str() const from /lib/x86_64-linux-gnu/libstdc++.so.6+0x4
#2  0x7fdd4139019c in App::Document::getName() const from /freecad_build/github2/FreeCAD/_build/lib/libFreeCADApp.so+0x1e
#3  0x7fdd41605a1e in App::ObjectIdentifier::resolve(App::ObjectIdentifier::ResolveResults&) const from /freecad_build/github2/FreeCAD/_build/lib/libFreeCADApp.so+0x114
#4  0x7fdd4160c60c in App::ObjectIdentifier::ResolveResults::ResolveResults(App::ObjectIdentifier const&) from /freecad_build/github2/FreeCAD/_build/lib/libFreeCADApp.so+0x1ec
#5  0x7fdd41608105 in App::ObjectIdentifier::getDocumentName() const from /freecad_build/github2/FreeCAD/_build/lib/libFreeCADApp.so+0x33
#6  0x7fdd416066cc in App::ObjectIdentifier::getDocument(App::ObjectIdentifier::String, bool*) const from /freecad_build/github2/FreeCAD/_build/lib/libFreeCADApp.so+0x52
#7  0x7fdd41606970 in App::ObjectIdentifier::getDocumentObject() const from /freecad_build/github2/FreeCAD/_build/lib/libFreeCADApp.so+0x6e
#8  0x7fdd4359baba in Gui::QuantitySpinBoxPrivate::parseString(QString const&, Base::Quantity&, double&, App::ObjectIdentifier const&) const from /freecad_build/github2/FreeCAD/_build/lib/libFreeCADGui.so+0xbe
#9  0x7fdd4359ccdc in Gui::QuantitySpinBoxPrivate::validateAndInterpret(QString&, QValidator::State&, App::ObjectIdentifier const&) const from /freecad_build/github2/FreeCAD/_build/lib/libFreeCADGui.so+0x1012
#10  0x7fdd4359a27b in Gui::QuantitySpinBox::validate(QString&, int&) const from /freecad_build/github2/FreeCAD/_build/lib/libFreeCADGui.so+0x57
#11  0x7fdd3f660868 in QWidgetLineControl::finishChange(int, bool, bool) from /lib/x86_64-linux-gnu/libQt5Widgets.so.5+0xf8
#12  0x7fdd3f660c2e in QWidgetLineControl::internalSetText(QString const&, int, bool) from /lib/x86_64-linux-gnu/libQt5Widgets.so.5+0x1ae
#13  0x7fdd43598031 in Gui::QuantitySpinBox::updateEdit(QString const&) from /freecad_build/github2/FreeCAD/_build/lib/libFreeCADGui.so+0xaf
#14  0x7fdd43597f43 in Gui::QuantitySpinBox::updateText(Base::Quantity const&) from /freecad_build/github2/FreeCAD/_build/lib/libFreeCADGui.so+0x7f
#15  0x7fdd43599271 in Gui::QuantitySpinBox::stepBy(int) from /freecad_build/github2/FreeCAD/_build/lib/libFreeCADGui.so+0xdf
#16  0x7fdd3f6b46e1 in QAbstractSpinBox::wheelEvent(QWheelEvent*) from /lib/x86_64-linux-gnu/libQt5Widgets.so.5+0x81
#17  0x7fdd3f551db8 in QWidget::event(QEvent*) from /lib/x86_64-linux-gnu/libQt5Widgets.so.5+0x1a8
#18  0x7fdd4359983d in Gui::QuantitySpinBox::event(QEvent*) from /freecad_build/github2/FreeCAD/_build/lib/libFreeCADGui.so+0x23
#19  0x7fdd3f50ffae in QApplicationPrivate::notify_helper(QObject*, QEvent*) from /lib/x86_64-linux-gnu/libQt5Widgets.so.5+0x7e
#20  0x7fdd3f518e06 in QApplication::notify(QObject*, QEvent*) from /lib/x86_64-linux-gnu/libQt5Widgets.so.5+0x1c76
#21  0x7fdd42eef000 in Gui::GUIApplication::notify(QObject*, QEvent*) from /freecad_build/github2/FreeCAD/_build/lib/libFreeCADGui.so+0x134
#22  0x7fdd3ea3b738 in QCoreApplication::notifyInternal2(QObject*, QEvent*) from /lib/x86_64-linux-gnu/libQt5Core.so.5+0x118
#23  /lib/x86_64-linux-gnu/libQt5Widgets.so.5(+0x1bf7ab) [0x7fdd3f56c7ab]
#24  /lib/x86_64-linux-gnu/libQt5Widgets.so.5(+0x1c1095) [0x7fdd3f56e095]
#25  0x7fdd3f50ffae in QApplicationPrivate::notify_helper(QObject*, QEvent*) from /lib/x86_64-linux-gnu/libQt5Widgets.so.5+0x7e
#26  0x7fdd42eef000 in Gui::GUIApplication::notify(QObject*, QEvent*) from /freecad_build/github2/FreeCAD/_build/lib/libFreeCADGui.so+0x134
#27  0x7fdd3ea3b738 in QCoreApplication::notifyInternal2(QObject*, QEvent*) from /lib/x86_64-linux-gnu/libQt5Core.so.5+0x118
#28  0x7fdd3ee16e72 in QGuiApplicationPrivate::processWheelEvent(QWindowSystemInterfacePrivate::WheelEvent*) from /lib/x86_64-linux-gnu/libQt5Gui.so.5+0xe2
#29  0x7fdd3edefcec in QWindowSystemInterface::sendWindowSystemEvents(QFlags<QEventLoop::ProcessEventsFlag>) from /lib/x86_64-linux-gnu/libQt5Gui.so.5+0xac
#30  /lib/x86_64-linux-gnu/libQt5XcbQpa.so.5(+0x6deca) [0x7fdd38327eca]
#31  /lib/x86_64-linux-gnu/libglib-2.0.so.0(g_main_context_dispatch+0x299) [0x7fdd3c8767a9]
#32  /lib/x86_64-linux-gnu/libglib-2.0.so.0(+0x54a38) [0x7fdd3c876a38]
#33  /lib/x86_64-linux-gnu/libglib-2.0.so.0(g_main_context_iteration+0x2c) [0x7fdd3c876acc]
#34  0x7fdd3ea93876 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) from /lib/x86_64-linux-gnu/libQt5Core.so.5+0x66
#35  0x7fdd3ea3a1bb in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) from /lib/x86_64-linux-gnu/libQt5Core.so.5+0x12b
#36  0x7fdd3ea42316 in QCoreApplication::exec() from /lib/x86_64-linux-gnu/libQt5Core.so.5+0x96
#37  /freecad_build/github2/FreeCAD/_build/lib/libFreeCADGui.so(+0x129ad91) [0x7fdd42dabd91]
#38  /freecad_build/github2/FreeCAD/_build/lib/libFreeCADGui.so(+0x129b055) [0x7fdd42dac055]
#39  0x7fdd42dac323 in Gui::Application::runApplication() from /freecad_build/github2/FreeCAD/_build/lib/libFreeCADGui.so+0x1d3
#40  ./bin/FreeCAD(+0x2d683) [0x559ae0063683]
#41  /lib/x86_64-linux-gnu/libc.so.6(+0x2724a) [0x7fdd3e35424a]
#42  /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0x85) [0x7fdd3e354305]
#43  ./bin/FreeCAD(+0x2c801) [0x559ae0062801]
`

or

`
Program received signal SIGSEGV, Segmentation fault.
#0  /lib/x86_64-linux-gnu/libc.so.6(+0x3c050) [0x7f67c8eee050]
#1  0x7f67c9203014 in std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::c_str() const from /lib/x86_64-linux-gnu/libstdc++.so.6+0x4
#2  0x7f67cbf1519c in App::Document::getName() const from /freecad_build/github2/FreeCAD/_build/lib/libFreeCADApp.so+0x1e
#3  0x7f67cc18aa1e in App::ObjectIdentifier::resolve(App::ObjectIdentifier::ResolveResults&) const from /freecad_build/github2/FreeCAD/_build/lib/libFreeCADApp.so+0x114
#4  0x7f67cc19160c in App::ObjectIdentifier::ResolveResults::ResolveResults(App::ObjectIdentifier const&) from /freecad_build/github2/FreeCAD/_build/lib/libFreeCADApp.so+0x1ec
#5  0x7f67cc18d105 in App::ObjectIdentifier::getDocumentName() const from /freecad_build/github2/FreeCAD/_build/lib/libFreeCADApp.so+0x33
#6  0x7f67cc18b6cc in App::ObjectIdentifier::getDocument(App::ObjectIdentifier::String, bool*) const from /freecad_build/github2/FreeCAD/_build/lib/libFreeCADApp.so+0x52
#7  0x7f67cc18b970 in App::ObjectIdentifier::getDocumentObject() const from /freecad_build/github2/FreeCAD/_build/lib/libFreeCADApp.so+0x6e
#8  0x7f67ce120aba in Gui::QuantitySpinBoxPrivate::parseString(QString const&, Base::Quantity&, double&, App::ObjectIdentifier const&) const from /freecad_build/github2/FreeCAD/_build/lib/libFreeCADGui.so+0xbe
#9  0x7f67ce121cdc in Gui::QuantitySpinBoxPrivate::validateAndInterpret(QString&, QValidator::State&, App::ObjectIdentifier const&) const from /freecad_build/github2/FreeCAD/_build/lib/libFreeCADGui.so+0x1012
#10  0x7f67ce11f27b in Gui::QuantitySpinBox::validate(QString&, int&) const from /freecad_build/github2/FreeCAD/_build/lib/libFreeCADGui.so+0x57
#11  0x7f67ca1e5868 in QWidgetLineControl::finishChange(int, bool, bool) from /lib/x86_64-linux-gnu/libQt5Widgets.so.5+0xf8
#12  0x7f67ca1e5c2e in QWidgetLineControl::internalSetText(QString const&, int, bool) from /lib/x86_64-linux-gnu/libQt5Widgets.so.5+0x1ae
#13  0x7f67ce11d031 in Gui::QuantitySpinBox::updateEdit(QString const&) from /freecad_build/github2/FreeCAD/_build/lib/libFreeCADGui.so+0xaf
#14  0x7f67ce11cf43 in Gui::QuantitySpinBox::updateText(Base::Quantity const&) from /freecad_build/github2/FreeCAD/_build/lib/libFreeCADGui.so+0x7f
#15  0x7f67ce11e271 in Gui::QuantitySpinBox::stepBy(int) from /freecad_build/github2/FreeCAD/_build/lib/libFreeCADGui.so+0xdf
#16  0x7f67ca2396e1 in QAbstractSpinBox::wheelEvent(QWheelEvent*) from /lib/x86_64-linux-gnu/libQt5Widgets.so.5+0x81
#17  0x7f67ca0d6db8 in QWidget::event(QEvent*) from /lib/x86_64-linux-gnu/libQt5Widgets.so.5+0x1a8
#18  0x7f67ce11e83d in Gui::QuantitySpinBox::event(QEvent*) from /freecad_build/github2/FreeCAD/_build/lib/libFreeCADGui.so+0x23
#19  0x7f67ca094fae in QApplicationPrivate::notify_helper(QObject*, QEvent*) from /lib/x86_64-linux-gnu/libQt5Widgets.so.5+0x7e
#20  0x7f67ca09de06 in QApplication::notify(QObject*, QEvent*) from /lib/x86_64-linux-gnu/libQt5Widgets.so.5+0x1c76
#21  0x7f67cda74000 in Gui::GUIApplication::notify(QObject*, QEvent*) from /freecad_build/github2/FreeCAD/_build/lib/libFreeCADGui.so+0x134
#22  0x7f67c95c0738 in QCoreApplication::notifyInternal2(QObject*, QEvent*) from /lib/x86_64-linux-gnu/libQt5Core.so.5+0x118
#23  /lib/x86_64-linux-gnu/libQt5Widgets.so.5(+0x1bf7ab) [0x7f67ca0f17ab]
#24  /lib/x86_64-linux-gnu/libQt5Widgets.so.5(+0x1c1095) [0x7f67ca0f3095]
#25  0x7f67ca094fae in QApplicationPrivate::notify_helper(QObject*, QEvent*) from /lib/x86_64-linux-gnu/libQt5Widgets.so.5+0x7e
#26  0x7f67cda74000 in Gui::GUIApplication::notify(QObject*, QEvent*) from /freecad_build/github2/FreeCAD/_build/lib/libFreeCADGui.so+0x134
#27  0x7f67c95c0738 in QCoreApplication::notifyInternal2(QObject*, QEvent*) from /lib/x86_64-linux-gnu/libQt5Core.so.5+0x118
#28  0x7f67c999be72 in QGuiApplicationPrivate::processWheelEvent(QWindowSystemInterfacePrivate::WheelEvent*) from /lib/x86_64-linux-gnu/libQt5Gui.so.5+0xe2
#29  0x7f67c9974cec in QWindowSystemInterface::sendWindowSystemEvents(QFlags<QEventLoop::ProcessEventsFlag>) from /lib/x86_64-linux-gnu/libQt5Gui.so.5+0xac
#30  /lib/x86_64-linux-gnu/libQt5XcbQpa.so.5(+0x6deca) [0x7f67c2eaceca]
#31  /lib/x86_64-linux-gnu/libglib-2.0.so.0(g_main_context_dispatch+0x299) [0x7f67c73fb7a9]
#32  /lib/x86_64-linux-gnu/libglib-2.0.so.0(+0x54a38) [0x7f67c73fba38]
#33  /lib/x86_64-linux-gnu/libglib-2.0.so.0(g_main_context_iteration+0x2c) [0x7f67c73fbacc]
#34  0x7f67c9618876 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) from /lib/x86_64-linux-gnu/libQt5Core.so.5+0x66
#35  0x7f67c95bf1bb in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) from /lib/x86_64-linux-gnu/libQt5Core.so.5+0x12b
#36  0x7f67c95c7316 in QCoreApplication::exec() from /lib/x86_64-linux-gnu/libQt5Core.so.5+0x96
#37  /freecad_build/github2/FreeCAD/_build/lib/libFreeCADGui.so(+0x129ad91) [0x7f67cd930d91]
#38  /freecad_build/github2/FreeCAD/_build/lib/libFreeCADGui.so(+0x129b055) [0x7f67cd931055]
#39  0x7f67cd931323 in Gui::Application::runApplication() from /freecad_build/github2/FreeCAD/_build/lib/libFreeCADGui.so+0x1d3
#40  ./bin/FreeCAD(+0x2d683) [0x5601139e0683]
#41  /lib/x86_64-linux-gnu/libc.so.6(+0x2724a) [0x7f67c8ed924a]
#42  /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0x85) [0x7f67c8ed9305]
#43  ./bin/FreeCAD(+0x2c801) [0x5601139df801]
`

Reply:
Sorry, but I don't have a space mouse device.